test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlServerFhirStorageTestsFixture.cs

```diff
@@ -145,7 +145,7 @@ internal SqlServerFhirStorageTestsFixture(int maximumSupportedSchemaVersion, str
                 NullLogger<SqlServerFhirDataStore>.Instance,
                 schemaInformation);
 
-            _fhirOperationDataStore = new SqlServerFhirOperationDataStore(SqlConnectionWrapperFactory, NullLogger<SqlServerFhirOperationDataStore>.Instance);
+            _fhirOperationDataStore = new SqlServerFhirOperationDataStore(SqlConnectionWrapperFactory, config, Substitute.For<IPollyRetryLoggerFactory>(), NullLogger<SqlServerFhirOperationDataStore>.Instance);
```

Comment:
If we're going to substitute for a `IPollyRetryLoggerFactory` in tests, we should add some assertions (like we called `Create()` n times). Otherwise we can keep it as the no-op version

Reply:
To summarize a bit of our discussion:

Here, we would be using a substitute of the `IPollyRetryLoggerFactory` as a dummy. Normally, substitutions involve assertions that check method invocation counts, etc. While using `NSubstitute` here is fine, it makes sense to investigate a way to distinguish scenarios where we are using `NSubstitute` to create dummy values from more traditional substitution use cases.